src/hooks/UseForm/JobApplication/JobApplication.tsx

```diff
@@ -0,0 +1,145 @@
+import React from 'react'
+import { Select } from '../../../storybook-common/components'
+import Button from '../../../storybook-common/components/Button'
+import useForm, { Errors } from '../useForm'
+
+enum Title {
+  Mr = 'MR',
+  Mrs = 'MRS',
+  Miss = 'MISS',
+  Ms = 'MS',
+}
+
+export type ApplicationFormData = {
+  email: string
+  title: Title
+  name: string
+  startDate: Date
+  expectedSalary: number
+  hasAgreedToTerms: boolean
+}
+
+export type ApplicationErrors = Errors<ApplicationFormData>
+
+interface Props {
+  onSubmit: (data: ApplicationFormData) => void
+  onError: (data: ApplicationErrors) => void
+}
+
+const JobApplication: React.FC<Props> = ({ onSubmit, onError }) => {
+  const { register, errors, hasBegun, hasChangedWithoutSubmit, handleSubmit } =
+    useForm<ApplicationFormData>()
+
+  return (
+    <div className="flex w-96 flex-col gap-4">
+      <div className="flex flex-col gap-1">
+        <label className="block text-sm font-medium text-gray-900">
+          Your email
+          <input
+            type="email"
+            className="block w-full rounded-lg border border-gray-300 bg-gray-50 p-2.5 text-sm text-gray-900 focus:border-blue-500 focus:ring-blue-500 "
+            placeholder="name@example.com"
+            {...register('email', {
+              isRequired: true,
+              validate: email => (email.includes('@') ? null : 'Email must include an @'),
+            })}
+          />
+        </label>
+        {errors?.email && <span className="text-red-800">{errors.email}</span>}
+      </div>
+      <div className="flex flex-col gap-1">
+        <Select
+          options={[
+            { label: 'Mr', value: Title.Mr },
+            { label: 'Mrs', value: Title.Mrs },
+            { label: 'Miss', value: Title.Miss },
+            { label: 'Ms', value: Title.Ms },
+          ]}
+          placeholder="Title"
+          {...register('title', { isRequired: true, refName: 'selectRef' })}
```

Comment:
I think having an entire component is overkill since all it does is remap key names and this is not a component library. I added a utility function that does the remapping and preserves all of our `register`'s types and generics, and wired it up for our `Select` in each of the form stories. Let me know if that does not seem like a sufficient solution. I tried adding the options to the `RegisterOptions` but ran into an issue where, because the new names are a dynamic prop, TS tries to infer the type of the new keys rather than preserving the defaults. It infers them as every single prop to whatever component `register` is called on, resulting in tons of type errors and warnings that the `register` is going to overwrite any given prop.

Reply:
Gotcha. I think that's fine, but just to clarify, if someone using this hook comes across a component that is missing any ref prop (so renaming won't fix it), are we not able to handle that or is it just that certain functionality is lost?

Reply:
The types for currently allow it but it could be updated to not require the ref. I believe the only functionality we would lose is picking up default values and converting the values to the proper types via the field's `type` attributes. Including `valueAsNumber` and `valueAsDate` should be easy enough if the ref is provided, just some additional checks in the onChange utils.

Reply:
Actually we currently prune fields that don't have a ref value before submission, so their fields would not be kept without refs currently. I think we could add a flag for if they have ever had a ref to distinguish fields that aren't set up that way from those that have been removed from the DOM since registering, so we should be able to support not setting the ref if we want.

Reply:
I went ahead and made it so users can disable auto-pruning for the entire form or per field. I also added functions to deregister fields one at a time or by coordinates. Fields can be registered without refs as long as `shouldNotBeAutoPruned` is set to false, users just won't be able to have default values in them and the returned data may not be parsed to the intended type. Also, they would have to manually remove those fields if they no longer applied. As for `valueAsNumber` and `valueAsDate` it seems that [React does not support them ](https://react.dev/reference/react-dom/components/input) so I don'tthink we need to handle them either.